azure/durable_functions/tasks/task_utilities.py

```diff
@@ -191,3 +192,208 @@ def set_processed(tasks):
     for task in tasks:
         if task is not None:
             task.is_processed = True
+
+
+def find_sub_orchestration(
+        state: List[HistoryEventType],
+        event_type: HistoryEventType,
+        name: Optional[str] = None,
+        counter: Optional[int] = None,
+        instance_id: Optional[str] = None,
+        scheduled_task: Optional[HistoryEvent] = None) -> Optional[HistoryEvent]:
+    """Look-up matching sub-orchestrator event in the state array.
+
+    Parameters
+    ----------
+    state: List[HistoryEventType]
+        The history of Durable events
+    event_type: HistoryEventType
+        The type of Durable event to look for.
+    name: str:
+        Name of the sub-orchestrator.
+    counter: int
+        The number of sub-orchestrations created before this
+        sub-orchestration was created.
+    instance_id: Optional[str], optional:
+        Instance ID of the sub-orchestrator. Defaults to None.
+    scheduled_task" Optional[HistoryEvent], optional:
+        The corresponding `scheduled` task for the searched-for event,
+        only available when looking for a completed or failed event.
+        Defaults to None.
+
+    Returns
+    -------
+    Optional[HistoryEvent]:
+        The matching event from the state array, if it exists.]
+    """
+
+    def gen_err_message(counter: int, mid_message: str, found: str, expected: str) -> str:
+        beg = f"The sub-orchestration call (n = {counter}) should be executed with "
+        middle = mid_message.format(found, expected)
+        end = " Check your code for non-deterministic behavior."
+        err_message = beg + middle + end
+        return err_message
+
+    event: Optional[HistoryEvent] = find_matching_event(state, None, event_type)
+
+    # Test for name and instance_id mistaches and, if so, error out.
+    # Also increase sub-orchestrator counter, for reporting.
+    if HistoryEventType.SUB_ORCHESTRATION_INSTANCE_CREATED:
```

Comment:
I'm a bit confused, because this doesn't seem like it is a valid boolean expression?

Reply:
Hmmm. yeah it appears that I accidentally pushed a slightly outdated file there. Fixed!